src/Runner.Worker/ActionRunner.cs

```diff
@@ -299,6 +298,44 @@ public bool TryEvaluateDisplayName(DictionaryContextData contextData, IExecution
             return didFullyEvaluate;
         }
 
+        private Dictionary<String, String> EvaluateStepInputs(
+            IStepHost stepHost
+        )
+        {
+            DictionaryContextData expressionValues;
+            if (stepHost is ContainerStepHost)
+            {
+                expressionValues = ExecutionContext.ExpressionValues.Clone() as DictionaryContextData;
+                UpdatePathsInExpressionValues("github", expressionValues, stepHost);
+                UpdatePathsInExpressionValues("runner", expressionValues, stepHost);
+            }
+            else
+            {
+                expressionValues = ExecutionContext.ExpressionValues;
+            }
+
+            // expression values of github = github dictionary
+            var templateEvaluator = ExecutionContext.ToPipelineTemplateEvaluator();
+            var inputs = templateEvaluator.EvaluateStepInputs(Action.Inputs, expressionValues, ExecutionContext.ExpressionFunctions);
+
+            return inputs;
+        }
+
+        private void UpdatePathsInExpressionValues(string contextName, DictionaryContextData expressionValues, IStepHost stepHost)
+        {
+            var dict = expressionValues[contextName].AssertDictionary($"expected context {contextName} to be a dictionary");
+            var output = dict.Clone().AssertDictionary($"expected dictionary clone in {contextName} to be dictionary");
```

Comment:
do we need to have `output`? can we update `dict` by looping through `dict.keys`?
ex:
```C#
foreach(var key in dict.Keys)
{
   var value = dict[key]?.ToString();
   if(!string.IsNullOrEmpty(value))
  {
     dict[key] = new StringContextData(stepHost.ResolvePathForStepHost(value));
  }
}
```